### DISCUSSION TITLE
javascript 태동기에 목표는 무엇이고, 지금과 어떤게 달라졌나요?

### DISCUSSION THREAD

**Discussion by greenblues1190** [woowacourse-study/2022-woowahan-bansanghwe#103] [JavaScript]
> Anyone interested in beta-testing a new client (Windows, Mac, or X versions), please send email to:
> beta@mcom.com
> Cheers, Marc
> — Marc Andreessen, announcing Mosaic (later Netscape) in [1994](http://1997.webhistory.org/www.lists/www-talk.1994q3/1121.html)

Javascript는 초기에 DOM을 동적으로 조작할 수 있는 **작은** 스크립팅 언어였습니다.

1995년 당시, 넷스케이프 커뮤니케이션즈의 설립자이자 Mosaic의 전 팀원이었던 Marc Andreessen은 웹이 보다 역동적으로 변할 방법이 필요하다는 비전을 가지고 있었습니다. 이 비전을 구체화할 언어에는 **소프트웨어 엔지니어링에 경험이 적은 사람들도 사용 가능할 것**이라는 요구사항이 있었습니다. Java의 부상과 함께 Java Applet(브라우저 상에서 applet 태그로 불러와 실행되는 응용프로그램)이 곧 현실화될 것이었기 때문에 디자이너와 같은 다른 유형의 사람들이 사용할 수 있을만큼 간단해야 했습니다. Brendan Eich는 **브라우저를 위한 Scheme**을 개발할 수 있다는 조건하에 넷스케이프에 합류했습니다. Scheme은 Lisp 방언이므로 구문적으로 매우 가벼웠고 동적이며, 강력하고, 함수형 언어였기에 넷스케이프가 요구하는 언어와 일치했습니다.

그러나 당시에 넷스케이프와 썬 마이크로시스템즈과의 협력이 성사되어 Java를 전문적인 프로그래머가 사용하도록하고 Mocha는 작은 스크립팅 작업에 사용하게 되었습니다. 브라우저를 위한 Scheme이었던 언어는 Java와 유사한 구문을 도입합니다. 그렇게 Java의 모습을 한 Scheme과 Self인 Javascript가 탄생합니다.

## 어떤게 달라졌나요?

- 브라우저마다 파편화된 구현을 표준화하기 위해 ECMAScript가 등장합니다. class, arrow function, const, let 등이 정의되었습니다.
- Node.js의 등장으로 서버 사이드 언어로도 사용가능해졌습니다.
- 타입스크립트가 등장했습니다.
- JIT 컴파일러를 내장한 크롬이 등장하며 자바스크립트는 더욱 강력한 언어가 되었습니다.

## 참고

- https://a-z.readthedocs.io/en/latest/javascript.html
- https://eager.io/blog/to-what-extent-did-marc-andreessen-invent-the-internet/
- https://auth0.com/blog/a-brief-history-of-javascript/
- https://www.cnet.com/tech/services-and-software/netscapes-andreessen-plugs-javascript/

**Comment by al-bur** [comment]
왜 프론트엔드 개발자??

**Comment by byhhh2** [comment]
프론트 측면이 빠른 변화가 있다면 반대의 공부는?

**Comment by byhhh2** [comment]
- 토이 프로젝트를 진행하면서 프론트가 잘 맞는다는 생각한 부분이 있는지

**Comment by byhhh2** [comment]
JS를 언제부터 사용했나요?

**Comment by byhhh2** [comment]
C언어와 JS언어의 차이점?

**Comment by byhhh2** [comment]
JS는 왜 컴파일 방식을 채택하지 않았을까?

**Comment by al-bur** [comment]
javascript 태동기에 목표는 무엇이고, 지금과 어떤게 달라졌나요?

**Comment by byhhh2** [comment]
- ECMAscript vs Javascript

**Comment by al-bur** [comment]
ES version에 따른 변화들

**Comment by al-bur** [comment]
자바스크립트를 사용하시면서 아쉬운 점이 있다면?

**Reply by byhhh2** [reply]
💌 To Javascript..
